third_party/terraform/resources/resource_google_billing_subaccount.go

```diff
@@ -0,0 +1,132 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"google.golang.org/api/cloudbilling/v1"
+)
+
+func resourceBillingSubaccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBillingSubaccountCreate,
+		Read:   resourceBillingSubaccountRead,
+		Delete: resourceBillingSubaccountDelete,
+		Update: resourceBillingSubaccountUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"display_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"open": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"master_billing_account": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"billing_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"rename_on_destroy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceBillingSubaccountCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	displayName := d.Get("display_name").(string)
+	masterBillingAccount := d.Get("master_billing_account").(string)
+
+	billingAccount := &cloudbilling.BillingAccount{
+		DisplayName:          displayName,
+		MasterBillingAccount: canonicalBillingAccountName(masterBillingAccount),
+	}
+
+	res, err := config.clientBilling.BillingAccounts.Create(billingAccount).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating billing subaccount '%s' in master account '%s': %s", displayName, masterBillingAccount, err)
+	}
+
+	d.SetId(res.Name)
+	d.Set("name", res.Name)
+	d.Set("open", res.Open)
+	d.Set("billing_account_id", GetResourceNameFromSelfLink(d.Get("name").(string)))
+
+	return nil
+}
+
+func resourceBillingSubaccountRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	id := d.Id()
+
+	res, err := config.clientBilling.BillingAccounts.Get(d.Id()).Do()
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("Billing Subaccount Not Found : %s", id))
+	}
+
+	billingAccount := res
```

Comment:
nit: you can just call this `billingAccount` in the first place on L82.

Reply:
fixed